aggregator_core/src/datastore.rs

```diff
@@ -3952,6 +3952,141 @@ impl<C: Clock> Transaction<'_, C> {
         .await?;
         Ok(())
     }
+
+    /// Retrieve all global HPKE keypairs.
+    #[tracing::instrument(skip(self), err)]
+    pub async fn get_global_hpke_keypairs(&self) -> Result<Vec<GlobalHpkeKeypair>, Error> {
+        let stmt = self
+            .prepare_cached(
+                "SELECT config_id, config, private_key, is_active, expired_at FROM global_hpke_keys;",
+            )
+            .await?;
+        let hpke_key_rows = self.query(&stmt, &[]).await?;
+
+        let mut global_hpke_keypairs = Vec::new();
+        for row in hpke_key_rows {
+            global_hpke_keypairs.push(self.global_hpke_keypair_from_row(&row)?)
+        }
+        Ok(global_hpke_keypairs)
+    }
+
+    /// Retrieve a global HPKE keypair by config ID.
+    #[tracing::instrument(skip(self), err)]
+    pub async fn get_global_hpke_keypair(
+        &self,
+        config_id: &HpkeConfigId,
+    ) -> Result<Option<GlobalHpkeKeypair>, Error> {
+        let stmt = self
+            .prepare_cached(
+                "SELECT config_id, config, private_key, is_active, expired_at FROM global_hpke_keys
+                    WHERE config_id = $1;",
+            )
+            .await?;
+        self.query_opt(&stmt, &[&(u8::from(*config_id) as i16)])
+            .await?
+            .map(|row| self.global_hpke_keypair_from_row(&row))
+            .transpose()
+    }
+
+    fn global_hpke_keypair_from_row(&self, row: &Row) -> Result<GlobalHpkeKeypair, Error> {
+        let config = HpkeConfig::get_decoded(row.get("config"))?;
+        let config_id = u8::try_from(row.get::<_, i16>("config_id"))?;
+
+        let encrypted_private_key: Vec<u8> = row.get("private_key");
+        let private_key = HpkePrivateKey::new(self.crypter.decrypt(
+            "global_hpke_keys",
+            &config_id.to_be_bytes(),
+            "private_key",
+            &encrypted_private_key,
+        )?);
+        Ok(GlobalHpkeKeypair::new(
+            HpkeKeypair::new(config, private_key),
+            row.get("is_active"),
+            row.get::<_, Option<NaiveDateTime>>("expired_at")
+                .as_ref()
+                .map(Time::from_naive_date_time),
+        ))
+    }
+
+    /// Unconditionally and fully drop a keypair. This is a dangerous operation,
+    /// since report shares encrypted with this key will no longer be decryptable.
+    #[tracing::instrument(skip(self), err)]
+    pub async fn delete_global_hpke_keypair(&self, config_id: &HpkeConfigId) -> Result<(), Error> {
```

Comment:
nit: it's totally fine to keep this since it's written & looks good, but in general it's a good idea to not implement functionality until you are ready to use it in application code.